web3/providers/rpc.py

```diff
@@ -57,6 +74,94 @@ def make_request(self, method, params):
         return response_body
 
 
+class KeepAliveRPCProvider(BaseProvider):
```

Comment:
I'd like to drop `RPCProvider` away if possible and just do `RPCProvider = KeepAliveRPCProvider` at the end of file. Then for `TestRPCProvider` do it directly from `BaseProvider`. I feel the classes are so small that chaining inheritance hierarchy at the moment will add more complexity with little benefit, as basically `make_request` is the only function.